yarnitor/background/worker.py

```diff
@@ -361,22 +520,42 @@ def run_update(self):
         # https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators
         self.state["refresh-datetime"] = datetime.datetime.utcnow().isoformat() + 'Z'
         self.redis_client.set(YARN_STATUS_KEY, json.dumps(self.state))
+        logger.info("Done updating metrics from YARN")
+
+    def loop(self, sleep_time):
+        """Executes the `run_update` message in a loop that catches and logs
+        all exceptions indefinitely.
 
-    def loop(self):
+        Parameters
+        ----------
+        sleep_time: float
+            Time to sleep after executing `run_update`
+        """
         while True:
             try:
                 self.run_update()
             except Exception:
                 logger.exception('Unknown exception while updating')
-            time.sleep(self.sleep_time)
+            time.sleep(sleep_time)
```

Comment:
should rewrite this using coroutines 😈

Reply:
http://s2.quickmeme.com/img/a9/a97b031533577743ad29e104e88a1f2f05a79b259e24d93118129073385446d3.jpg